sympy/sets/sets.py

```diff
@@ -867,6 +891,13 @@ def _measure(self):
             measure *= s.measure
         return measure
 
+    @property
+    def _kind(self):
+        if self.args[0].kind == self.args[1].kind:
+            return self.args[0].kind
```

Comment:
For ProductSet like `Reals*Integers` the set is a set of tuples:
```python
In [11]: S = (Reals*Integers)

In [12]: S
Out[12]: ℝ × ℤ

In [13]: (1.5, 2) in S
Out[13]: True
```
The kind should be in this case `SetKind(TupleKind(NumberKind, NumberKind)))`.

Also this should be fixed:
```python
In [14]: Tuple(1, 2).kind
Out[14]: UndefinedKind
```

Reply:
```python
>>> Tuple(1,2).kind
TupleKind(NumberKind, NumberKind)
```
for powerset
A = (1,2)
B = (3,4)
P = A * B = ([1,3], [1,4], [2,3], [2,4])
so this is tuple of 4 set?
```python
>>> ProductSet(Inteval(1,2), Interval(3,4)).kind
TupleKind(SetKind(NumberKind))
```

Reply:
Should Tuple be something like this
```python
>>> Tuple(1, 2).kind
TupleKind(NumberKind)
>>> Tuple(1, Matrix([1, 2])).kind
TupleKind(UndefinedKind)
```
For PowerSet since it is set of tuples
```python
>>>PowerSet(Interval(1, 2), Interval(5, 6)).kind
SetKind(TupleKind(NumberKind))
>>>
```

Reply:
Tuples can have mixed kinds so the kind of each element should be specified e.g.:
```python
>>> Tuple(1, Matrix([1, 2])).kind
TupleKind(NumberKind, MatrixKind(NumberKind))
```
PowerSet would be:
```python
>>> PowerSet({1,2,3}).kind
SetKind(SetKind(NumberKind))
```
ProductSet would be:
```python
>>> ProductSet(FiniteSet(1,Matrix([1, 2])), Interval(1, 2))
SetKind(TupleKind(UndefinedKind, NumberKind))
```